### DISCUSSION TITLE
Node mass output issue

### DISCUSSION THREAD

**Discussion by Aaboe-KU** [OpenRadioss/OpenRadioss#3143] [Q&A]
Hello everyone,
I am doing a relatively simple collision simulation, for a project, using OpenRadioss and I need to be able to output the mass of each node to my vtk-files, after the simulation. The user guide simply says to run the /ANIM/MASS command: https://help.altair.com/hwsolvers/rad/topics/solvers/rad/anim_mass_engine_r.htm

However, when I do this, I get no mass output whatsoever. My current output file contains the following:

**/RUN/Impact_1/1
1200e-6
/VERS/2024
/DT
0.9 1e-11
/ANIM/DT
0.0 10e-06
/ANIM/ELEM/VONM
/ANIM/VECT/DISP
/ANIM/VECT/VEL
/ANIM/VECT/VROT
/ANIM/VECT/ACC
/ANIM/VECT/CONT
/ANIM/VECT/PCONT
/ANIM/GPS/STRESS/TENS
/ANIM/GPS/STRAIN/TENS
/ANIM/ELEM/VOLU
/ANIM/ELEM/DENS
/ANIM/NODA/DENS
/ANIM/MASS**

All of the other outputs are nicely contained in the resulting vtk-files after the simulation. All except the actual node masses. I have attempted multiple things, but nothing has worked so far. I can see in my .out files, that my outputs seem correct. It even tells me that the nodal masses should be outputting:

**VARIABLE SAVED ON ANIMATION FILES (1: YES) :
PLASTIC STRAIN. . . . . .     0
DENSITY     (SOLID ONLY).     1
SPECIFIC ENERGY . . . . .     0
HOURGLASS ENERGY. . . . .     0
TEMPERATURE (SOLID ONLY).     0
THICKNESS   (SHELL ONLY).     0
PRESSURE    (SOLID ONLY).     0
VON MISES . . . . . . . .     1
TURBULENT ENERGY (FLUID).     0
TURBULENT VISCOSITY(FLUID)    0
VORTICITY (FLUID) . . . .     0
NODAL VALUES SAVED ON ANIMATION FILES (1: YES) :
PRESSURE. . . . . . . . .     0
DENSITY . . . . . . . . .     1
TEMPERATURE . . . . . . .     0
VECTOR SAVED ON ANIMATION FILES (1: YES) :
VELOCITY VECTOR . . . . .     1
DISPLACEMENT VECTOR . . .     1
ACCELERATION VECTOR . . .     1
CONTACT FORCES. . . . . .     1
INTERNAL FORCES . . . . .     0
EXTERNAL FORCES . . . . .     0
SECTION RBY RWALL FORCES.     0
ROTATIONAL VELOCITY VECTOR    1
CONTACT PRESSURE (VECTORS)    1
SHELL TENSOR SAVED ON ANIMATION FILES (1: YES) :
MEMBRANE STRESS . . . . .     0
BENDING STRESS(MOMENT/T^2)    0
UPPER LAYER STRESS. . . .     0
LOWER LAYER STRESS. . . .     0
MEMBRANE STRAIN . . . . .     0
CURVATURE . . . . . . . .     0
UPPER LAYER STRAIN. . . .     0
LOWER LAYER STRAIN. . . .     0
NODAL MASS SAVED ON ANIMATION FILES (1: YES) :    1
KEEP DELETED ELEMENT (1: YES) :    0**

Strangely though, it also tells me:
**NODMAS: NUMBER OF ADDED NODAL MASSES. . . . . . . . .         0**
And:
**NODAL MASS DISTRIBUTION FLAG . . . . . . . . .         0**
I am still quite novice at these things, so if anyone has any suggestions, solutions or advice, it'd be much appreciated.

**Comment by victorkemp** [comment]
Nodal mass is written to a special part of the .Annn files, different from general nodal scalars. I just tried it and can see values in there. However, from looking at the anim_to_vtk source code, the converter appears to read them but not write them to vtk. It probably wouldn't be hard to add that capability and compile it - it's just a single file c++ program with no special dependencies as far as I can tell.
